src/Application.vala

```diff
@@ -56,6 +56,27 @@ public class Terminal.Application : Gtk.Application {
         Intl.bindtextdomain (Config.GETTEXT_PACKAGE, Config.LOCALEDIR);
         Intl.bind_textdomain_codeset (Config.GETTEXT_PACKAGE, "UTF-8");
         Intl.textdomain (Config.GETTEXT_PACKAGE);
+
+        var act = new SimpleAction ("process-finished", VariantType.STRING);
+        add_action (act);
+        act.activate.connect ((v) => {
+            var now = new DateTime.now_local ();
+            size_t len;
+            var tid = v.get_string (out len);
+            foreach (var window in windows) {
+                foreach (var terminal in window.terminals) {
+                    if (terminal.terminal_id == tid) {
+                        window.set_active_terminal_tab (terminal.tab);
+                        var ts = Gdk.X11.get_server_time ((Gdk.X11.Window) window.get_window ());
+                        window.present_with_time (ts);
```

Comment:
should be using `Gtk.get_current_event_time ()`
```suggestion
                        window.present_with_time (Gtk.get_current_event_time ());
```

Reply:
@Marukesu Does this work on your end? It doesn't for me. From what I've read, the  timestamp that `present_with_time` receives must be a X11 server timestamp and the only way I was able to get it was using this Gdk.X11 call. I've tried `Gdk.CURRENT_TIME` and many types of datetime functions returning the current time as well. 

I'm not a fan of adding this dependency too, but it was literally the only timestamp that worked for me.

Reply:
I can confirm that `current_event_time` does not work on X11 or Wayland, probably because the event did not occur on the Terminal window but was received over DBus.  I have suggested a hack to get it work on X11 at least with that extra dependency.